ts/session/utils/InsecureNodeFetch.ts

```diff
@@ -91,16 +93,201 @@ type NodeFetchParams = {
+  const settings = getProxySettings();
+  if (!settings) {
+    window?.log?.debug('getProxyAgent: No proxy settings configured');
+    return undefined;
+  }
+
+  const auth =
+    settings.username && settings.password
+      ? `${encodeURIComponent(settings.username)}:${encodeURIComponent(settings.password)}@`
+      : '';
+  const proxyUrl = `socks5h://${auth}${settings.host}:${settings.port}`;
+
+  // Create cache key that includes TLS options
+  const useTlsOptions = hasTlsOptions(tlsOptions);
+  const tlsOptionsKey = useTlsOptions
+    ? JSON.stringify(Object.keys(tlsOptions).sort())
+    : 'no-tls';
+  const cacheKey = `${proxyUrl}:${tlsOptionsKey}`;
+
+  // Check cache to reuse existing agents (performance optimization)
+  const cachedAgent = cachedAgents.get(cacheKey);
+  if (cachedAgent) {
+    return cachedAgent;
+  }
+
+  // Create new agent with SOCKS5 configuration
+  window?.log?.info(`getProxyAgent: Creating new SOCKS5 agent for ${settings.host}:${settings.port}`);
+  if (useTlsOptions) {
+    window?.log?.debug(
+      `getProxyAgent: Applying TLS options to SOCKS agent: ${Object.keys(tlsOptions).join(', ')}`
+    );
+  }
+  const agent = new SocksProxyAgentWithTls(
+    proxyUrl,
+    {
+      timeout: 30000, // 30 seconds timeout for SOCKS connection
+    },
+    useTlsOptions ? tlsOptions : undefined
+  );
+
+  // Cache the agent
+  cachedAgents.set(cacheKey, agent);
+
+  return agent;
+}
+
 export async function insecureNodeFetch(params: NodeFetchParams) {
   try {
+    // Extract TLS options from the original agent (if present) to preserve security settings
+    // This ensures certificate pinning and other TLS configurations work through the proxy
+    const tlsOptions = getTlsOptionsFromAgent(params.fetchOptions?.agent);
+    const proxyAgent = getProxyAgent(tlsOptions);
+
+    // CRITICAL: Proxy agent must override any other agent (like sslAgent) to route through SOCKS
+    const fetchOptions = {
+      ...params.fetchOptions,
+      ...(proxyAgent ? { agent: proxyAgent } : {}),
+    } as RequestInit;
+
+    if (proxyAgent) {
+      window?.log?.info(`insecureNodeFetch: Using proxy for request to ${params.url}`);
+    }
+
     debugLogRequestIfEnabled(params);
-    const result = await nodeFetch(params.url, params.fetchOptions);
+    const result = await nodeFetch(params.url, fetchOptions);
     handleGoOnline(result, params.destination);
     debugLogResponseIfEnabled(result);
     return result;
   } catch (e) {
     handleGoOffline(e);
-    window?.log?.error('insecureNodeFetch', e);
+    // Enhanced error logging for debugging proxy issues
+    const errorDetails = {
+      message: (e as Error).message,
+      name: (e as Error).name,
+      code: (e as any).code,
+      errno: (e as any).errno,
+      syscall: (e as any).syscall,
+      type: (e as any).type,
+      stack: (e as Error).stack?.split('\n').slice(0, 3).join('\n'),
+    };
+    window?.log?.error('insecureNodeFetch error', errorDetails);
```

Comment:
this is going to be very verbose.
Maybe split it with two debug levels:
one `window?.log?.error` with just the `insecureNodeFetch error: ${e.message}` and one with window?.log?.debug with what you have currently